cla_backend/apps/knowledgebase/tests/test_csv_2_fixture.py

```diff
@@ -28,1416 +47,724 @@ def slice_records(self, output_records, expected_records, start_index=None, end_
-                    u"preferred_signpost": False,
-                },
-            },
-            {
-                u"pk": 33,
-                u"model": u"knowledgebase.articlecategorymatrix",
-                u"fields": {
-                    u"created": u"{date}".format(date=self.datetime_now),
-                    u"modified": u"{date}".format(date=self.datetime_now),
-                    u"article": 2,
-                    u"article_category": 15,
-                    u"preferred_signpost": False,
-                },
-            },
-            {
-                u"pk": 34,
-                u"model": u"knowledgebase.articlecategorymatrix",
-                u"fields": {
-                    u"created": u"{date}".format(date=self.datetime_now),
-                    u"modified": u"{date}".format(date=self.datetime_now),
-                    u"article": 2,
-                    u"article_category": 16,
-                    u"preferred_signpost": False,
-                },
-            },
-            {
-                u"pk": 35,
-                u"model": u"knowledgebase.articlecategorymatrix",
-                u"fields": {
-                    u"created": u"{date}".format(date=self.datetime_now),
-                    u"modified": u"{date}".format(date=self.datetime_now),
-                    u"article": 2,
-                    u"article_category": 17,
-                    u"preferred_signpost": True,
-                },
-            },
-            {
-                u"pk": 36,
-                u"model": u"knowledgebase.articlecategorymatrix",
-                u"fields": {
-                    u"created": u"{date}".format(date=self.datetime_now),
-                    u"modified": u"{date}".format(date=self.datetime_now),
-                    u"article": 2,
-                    u"article_category": 18,
-                    u"preferred_signpost": True,
-                },
-            },
-        ]
-        outputList = json.loads(outputJSON)
+    def test_fixture_handling_with_entry_type_of_other_resource_for_clients(self):
+        csv_file = self.find_path_to_csvfile("csv_with_entry_type")
+        file = open(csv_file)
+        try:
+            csv = KnowledgebaseCsvParse(file)
+        except SystemExit:
+            self.fail("Missing required fields")
+        else:
+            outputJSON = csv.fixture_as_json()
+            outputList = json.loads(outputJSON)
+            outputArticle = outputList[-1]
+            self.assertEqual(outputArticle["fields"]["resource_type"], u"OTHER")
+
+    def test_fixture_with_complete_article(self):
+        csv_file = self.find_path_to_csvfile("complete_article")
+        file = open(csv_file)
+        try:
+            csv = KnowledgebaseCsvParse(file)
+        except SystemExit:
+            self.fail("Missing required fields")
+        else:
+            outputJSON = csv.fixture_as_json()
+            outputList = json.loads(outputJSON)
+            expectedList = [
```

Comment:
I would probably prefer a check of just the attributes you care about, for previously-discussed reasons of communicating intent. I wonder if you could restructure this as either looping through the output and asserting the values you want to assert or -- and this second one is probably preferable -- converting expectedList into a list of just the values you want to check (probably `article`, `article_category` and `preferred_signpost`) and then you can assert that that list matches a defined output (which would then be much easier to read than our current expected output)